src/fec_rs_gf2n_fft_add.h

```diff
@@ -183,18 +186,18 @@ class RsGf2nFftAdd : public FecCode<T> {
     }
 
     void decode_apply(
-        DecodeContext<T>* context,
+        const DecodeContext<T>& context,
         vec::Vector<T>* output,
         vec::Vector<T>* words) override
     {
-        vec::Vector<T>* fragments_ids = context->fragments_ids;
-        std::shared_ptr<vec::Poly<T>> A = context->A;
-        std::shared_ptr<vec::Poly<T>> inv_A_i = context->inv_A_i;
-        std::shared_ptr<vec::Poly<T>> poly1_n = context->poly1_n;
-        std::shared_ptr<vec::Poly<T>> S = context->S;
+        vec::Vector<T>* fragments_ids = context.get_fragments_id();
```

Comment:
ditto, better if these are reference instead of pointers.

Also, the accessor could probably named `vec` and `poly`.

i.e
```C++
        vec::Poly<T>& A = context.poly(CtxPoly::A);
        vec::Vector<T>& inv_A_i = context.vector(CtxVec::INV_A_I);
        vec::Vector<T>& vec1_n = context.vec(CtxVec::N1);
        vec::Poly<T>& S = context.poly(CtxPoly::S);
```